conn.go

```diff
@@ -182,14 +196,29 @@ func (conn *Conn) Close() error {
 		conn.outLck.Unlock()
 		return nil
 	}
+
 	close(conn.out)
 	conn.closed = true
 	conn.outLck.Unlock()
+
+	close(conn.shutdownSigWorker)
+	conn.signalQueueLck.Lock()
+	// This is to avoid unnecessary wakeup of sigWorker
+	tmp := conn.signalQueueWakeUp
+	conn.signalQueueWakeUp = nil
+	close(tmp)
+	conn.signalQueueLck.Unlock()
+
 	conn.signalsLck.Lock()
+	// This is to avoid closing channels user decided to remove
+	// from signal channels list
+	conn.processPendingSignalChannels()
 	for _, ch := range conn.signals {
 		close(ch)
 	}
+	conn.signals = nil
```

Comment:
It hurts in the sense that people are going to ask this question every time they see this line. Let's just remove the code if it's not necessary.